src/Adapter/Customer/QueryHandler/GetCustomerInformationHandler.php

```diff
@@ -0,0 +1,590 @@
+        }
+
+        $registrationDate = Tools::displayDate($customer->date_add, null, true);
+        $lastUpdateDate = Tools::displayDate($customer->date_upd, null, true);
+        $lastVisitDate = $customerStats['last_visit'] ?
+            Tools::displayDate($customerStats['last_visit'], null, true) :
+            $this->translator->trans('Never', [], 'Admin.Global')
+        ;
+
+        $customerShop = new Shop($customer->id_shop);
+        $customerLanguage = new Language($customer->id_lang);
+
+        $customerSubscriptions = new Subscriptions(
+            (bool) $customer->newsletter,
+            (bool) $customer->optin
+        );
+
+        return new PersonalInformation(
+            $customer->firstname,
+            $customer->lastname,
+            $customer->email,
+            $customer->isGuest(),
+            $socialTitle,
+            $birthday,
+            $registrationDate,
+            $lastUpdateDate,
+            $lastVisitDate,
+            $this->getCustomerRankBySales($customer->id),
+            $customerShop->name,
+            $customerLanguage->name,
+            $customerSubscriptions,
+            (bool) $customer->active
+        );
+    }
+
+    /**
+     * @param int $customerId
+     *
+     * @return int|null customer rank or null if customer is not ranked
+     */
+    private function getCustomerRankBySales($customerId)
+    {
+        $sql = 'SELECT SUM(total_paid_real) FROM ' . _DB_PREFIX_ . 'orders WHERE id_customer = ' . (int) $customerId . ' AND valid = 1';
+
+        if ($totalPaid = Db::getInstance()->getValue($sql)) {
+            $sql = '
+                SELECT SQL_CALC_FOUND_ROWS COUNT(*)
+                FROM ' . _DB_PREFIX_ . 'orders
+                WHERE valid = 1
+                    AND id_customer != ' . (int) $customerId . '
+                GROUP BY id_customer
+                HAVING SUM(total_paid_real) > ' . (int) $totalPaid
+            ;
+
+            Db::getInstance()->getValue($sql);
+
+            return (int) Db::getInstance()->getValue('SELECT FOUND_ROWS()') + 1;
+        }
+
+        return null;
+    }
+
+    /**
+     * @param Customer $customer
+     *
+     * @return OrdersInformation
+     */
+    private function getCustomerOrders(Customer $customer)
+    {
+        $validOrders = [];
+        $invalidOrders = [];
+
+        $orders = Order::getCustomerOrders($customer->id, true);
+        $totalSpent = 0;
+
+        foreach ($orders as $order) {
+            $order['total_paid_real_not_formated'] = $order['total_paid_real'];
+            $order['total_paid_real'] = Tools::displayPrice(
+                $order['total_paid_real'],
+                new Currency((int) $order['id_currency'])
+            );
+
+            if (!isset($order['order_state'])) {
+                $order['order_state'] = $this->translator->trans(
+                    'There is no status defined for this order.',
```

Comment:
Should this be here or in the Twig template ? Is it really an `order_state` property of the Order, or is it a display rule only for templating ?

Reply:
i think its for templating, but since query `GetCustomerInformation` is all about getting view data, maybe its ok? :thinking: if you think no, ill update. :)

Reply:
I think this will be better in the template